### DISCUSSION TITLE
Lambda Best Practices

### DISCUSSION THREAD

**Discussion by coogle** [protomaps/PMTiles#308] [General]
Hello,

I am trying to formalize setting up Protomaps via AWS / Serverless using Lambda. The instructions point me to this URL to get the Javascript Bundle for the Lambda function:

https://protomaps.github.io/PMTiles/lambda_function.zip

That's all well and good, but is there a more "official" way to get this package? Can I build it myself or find it in a repository somewhere? The context here is trying to build Terraform resources / infrastructure-as-code... One option of course is to download the zip and bundle it with my IaC -- but then of course I have to know when to update it.

**Comment by bdon** [comment]
the script to build the lambda package is here: https://github.com/protomaps/PMTiles/blob/main/serverless/aws/package.json#L14

Is that aligned with your way of doing the Terraform deploy? I know of some users using that, and some using AWS CDK.

**Comment by coogle** [comment]
I'm not familiar personally with CDK so I can't comment there... but I suppose one option that comes to mind for TF here would be a bit complicated but functional.. Basically configure terraform to load the lambda payload from an S3 bucket, and then use that S3 bucket to provide updates via your build process's distribution output.

For now I'm cheating and just putting the JavaScript payload in my terraform directly as I described before simply because I don't have the time to go the more convoluted route.
